src/DocumentationBrowserViewExtension/DocumentationBrowserViewExtension.cs

```diff
@@ -132,8 +170,178 @@ public override void Loaded(ViewLoadedParams viewLoadedParams)
             {
                 OnPackageLoaded(pkg);
             }
+
+            this.DynamoViewModel = (viewLoadedParams.DynamoWindow.DataContext as DynamoViewModel);
+        }
+
+        private void OnInsertFile(object sender, InsertDocumentationLinkEventArgs e)
+        {
+            if (DynamoViewModel.Model.CurrentWorkspace is HomeWorkspaceModel)
+            {
+                var homeWorkspace = DynamoViewModel.Model.CurrentWorkspace as HomeWorkspaceModel;
+                if (homeWorkspace != null && homeWorkspace.RunSettings.RunType != RunType.Manual)
+                {
+                    DynamoViewModel.MainGuideManager.CreateRealTimeInfoWindow(Resources.ToastInsertGraphNotificationText, true);
+                    homeWorkspace.RunSettings.RunType = RunType.Manual;
+                }
+            }
+            
+            var existingGroups = GetExistingGroups();
+
+            // Insert the file and select all the elements that were inserted 
+            this.DynamoViewModel.Model.InsertFileFromPath(e.Data);
+
+            if (!DynamoSelection.Instance.Selection.Any()) return;
+            DoEvents();
+
+            GroupInsertedGraph(existingGroups, e.Name);
+            DoEvents();
+
+            // We have selected all the nodes and notes from the inserted graph
+            // Now is the time to auto layout the inserted nodes
+            this.DynamoViewModel.GraphAutoLayoutCommand.Execute(null);
+            DoEvents();
+
+            this.DynamoViewModel.FitViewCommand.Execute(null);
+            DoEvents();
         }
 
+
+        private void GroupInsertedGraph(List<AnnotationViewModel> existingGroups, string graphName)
+        {
+            var selection = GetCurrentSelection();
+            var hostGroups = GetAllHostingGroups(existingGroups);
+
+            foreach (var group in hostGroups)
+            {
+                group.DissolveNestedGroupsCommand.Execute(null);
+            }
+
+           
+            foreach (var group in hostGroups)
+            {
+                selection.RemoveAll(x => group.AnnotationModel.ContainsModel(x as ModelBase));
+            }
+
+            DynamoSelection.Instance.Selection.AddRange(selection);
+            DynamoSelection.Instance.Selection.AddRange(hostGroups.Select(x => x.AnnotationModel));
+            DoEvents();
+
+            var annotation = this.DynamoViewModel.Model.CurrentWorkspace.AddAnnotation(Resources.InsertedGroupSubTitle, Guid.NewGuid());
+            annotation.AnnotationText = graphName;
+            DoEvents();
+
+            var annotationViewModel = DynamoViewModel.CurrentSpaceViewModel.Annotations
+                    .First(x => x.AnnotationModel == annotation);
+
+            // Is that the correct way of getting a GroupStyleItem from StyleItem?
+            var styleItem = annotationViewModel.GroupStyleList.First(x => x.Name.Equals("Review"));
```

Comment:
Why is the group style name is "Review"? This doesn't seem correct.

Reply:
This was a comment by @Amoursol - when inserting an Example graph, we want to group the incoming graph inside one group and assign the color scheme for the 'Review' group style. Would that answer the question?

Reply:
Used the localized string from Dynamo.Properties! Thank you!!